pkg/util/steps/condition.go

```diff
@@ -66,9 +70,48 @@ func (c conditionStep) run(ctx context.Context, log *logrus.Entry) error {
 		log.Warnf("step %s failed but has configured 'fail=%t'. Continuing. Error: %s", c, c.fail, err.Error())
 		return nil
 	}
+	if err == wait.ErrWaitTimeout {
+		return c.enrichConditionTimeoutError()
+	}
 	return err
 }
 
+func (c conditionStep) enrichConditionTimeoutError() error {
+	funcNameParts := strings.Split(FriendlyName(c.f), ".")
+	funcName := strings.Replace(funcNameParts[len(funcNameParts)-1], "-fm", "", 1)
+
+	message := "timed out polling for a prerequisite cluster condition to resolve"
+	switch funcName {
+	case "apiServersReady":
+		message = "Kube API has not initialised successfully and is unavailable"
+	case "minimumWorkerNodesReady":
+		message = "minimum number of worker nodes have not been successfully created"
+	case "operatorConsoleExists":
+		message = "Console Cluster Operator has failed to initialize successfully"
+	case "operatorConsoleReady":
+		message = "Console Cluster Operator has not started successfully"
+	case "clusterVersionReady":
+		message = "Cluster Verion is not reporting status as ready"
+	case "ingressControllerReady":
+		message = "Ingress Cluster Operator has not started successfully"
+	case "aroDeploymentReady":
+		message = "ARO Cluster Operator has failed to initialize successfully"
+	case "ensureAROOperatorRunningDesiredVersion":
+		message = "ARO Cluster Operator is not running desired version"
+	case "hiveClusterDeploymentReady":
+		message = "Cluster Hive deployment is unsuccessful and is unavailable"
+	case "hiveClusterInstallationComplete":
+		message = "Cluster Hive Installation is unsuccessful"
+	default:
+		message = "timed out polling for a prerequisite cluster condition to resolve"
+	}
+	return api.NewCloudError(
+		http.StatusFailedDependency,
```

Comment:
Hmm... IDK about code 424 here - this indicates a client-side error, that being, that the customer previously submitted a request to our API that caused a subsequent one (this one) to fail - this is likely not the case. IMO we should be returning a 500 Internal Server Error for these, but open to discussion from the team.

Reply:
Made the change.